### PR TITLE
Adding python-botocore Spec in CBL-Mariner

### PR DESCRIPTION
**Pull request by Sumynwa**
(a) Initial import from fedora 36.
(b) Adding as dependency for cassandra medusa.
(c) License verified.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
Adding python-botocore spec in CBL-Mariner as dependency for cassandra medusa.

###### Change Log  
(a) Initial import from fedora 36.
(b) Adding as dependency for cassandra medusa.
(c) License verified.

###### Does this affect the toolchain?  
NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Local Build

### CODE REVIEW

File: SPECS/python-botocore/python-botocore.spec [hunk 1]
```diff
@@ -0,0 +1,1860 @@
+Summary:        Low-level, data-driven core of boto 3
+Name:           python-%{pypi_name}
+# NOTICE - Updating this package requires updating python-boto3
+Version:        1.24.33
```
Comment (oliviacrain) [inline SPECS/python-botocore/python-botocore.spec:4]: Can we update this to a newer version?

File: SPECS/python-botocore/python-botocore.spec [hunk 2]
```diff
@@ -0,0 +1,1860 @@
...
+%global pypi_name botocore
+# Replace removed inspect.formatargspec with inspect.signature
+# This is needed for compatibility with Python 3.11 and
+# can be removed once upstream PR is merged.
+Patch2507:      https://github.com/boto/botocore/pull/2507.patch
+BuildArch:      noarch
+
+BuildRequires:  pyproject-rpm-macros
+BuildRequires:  python3
+BuildRequires:  python3-pip
```
Comment (oliviacrain) [inline SPECS/python-botocore/python-botocore.spec:20]: Pip shouldn't be necessary for non-check builds, please gate appropriately

File: SPECS/python-botocore/python-botocore.spec [hunk 3]
```diff
@@ -0,0 +1,1860 @@
...
+Name:           python-%{pypi_name}
+# NOTICE - Updating this package requires updating python-boto3
+Version:        1.24.33
+Release:        2%{?dist}
+License:        ASL 2.0
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://github.com/boto/botocore
+Source0:        https://github.com/boto/botocore/archive/refs/tags/%{version}.tar.gz#/%{pypi_name}-%{version}.tar.gz
+%global pypi_name botocore
```
Comment (oliviacrain) [inline SPECS/python-botocore/python-botocore.spec:11]: Generally, this definition should be at the top of the file.

File: SPECS/python-botocore/python-botocore.spec [hunk 4]
```diff
@@ -0,0 +1,1860 @@
...
+# Remove online tests
+rm -vr tests/integration
+# This test tried to import tests/cmd-runner which failed as the code was
+# unable to import "botocore". I'm not 100% sure why this happened but for now
+# just exclude this one test and run all the other functional tests.
+rm -vr tests/functional/leak
+
+%generate_buildrequires
+# -r use final runtime dependencies as BuildRequires
+%pyproject_buildrequires -r
```
Comment (oliviacrain) [inline SPECS/python-botocore/python-botocore.spec:58]: We don't support this, please generate all requirements manually.

File: SPECS/python-botocore/python-botocore.spec [hunk 5]
```diff
@@ -0,0 +1,1860 @@
...
+
+%build
+%pyproject_wheel
+
+%install
+%pyproject_install
+
+%check
+%pytest3
```
Comment (oliviacrain) [inline SPECS/python-botocore/python-botocore.spec:67]: I don't think we have this macro in Mariner.

**Review by oliviacrain — CHANGES_REQUESTED** [review]

File: SPECS/python-botocore/python-botocore.spec (hunk 1 above)
Comment (Sumynwa) [reply SPECS/python-botocore/python-botocore.spec]: Updated to 1.27.35.

File: SPECS/python-botocore/python-botocore.spec (hunk 2 above)
Comment (Sumynwa) [reply SPECS/python-botocore/python-botocore.spec]: Done

File: SPECS/python-botocore/python-botocore.spec (hunk 3 above)
Comment (Sumynwa) [reply SPECS/python-botocore/python-botocore.spec]: Moved all %global to top of file.

File: SPECS/python-botocore/python-botocore.spec (hunk 4 above)
Comment (Sumynwa) [reply SPECS/python-botocore/python-botocore.spec]: Done

File: SPECS/python-botocore/python-botocore.spec (hunk 5 above)
Comment (Sumynwa) [reply SPECS/python-botocore/python-botocore.spec]: This is a typo. Corrected.

File: SPECS/python-botocore/2507.patch [hunk 6]
```diff
@@ -0,0 +1,29 @@
+From ec30ecf89f8788da139f73d9140de66fc21848d5 Mon Sep 17 00:00:00 2001
```
Comment (oliviacrain) [inline SPECS/python-botocore/2507.patch:1]: Please delete this file, since we no longer use the patch.

**Review by oliviacrain — APPROVED** [review]
Looks good to me, other than the below comment- please address before merging

**Pull request merged**